### PR TITLE
CCIBI-3739: Input component improvements

### PR DESCRIPTION
**Pull request by andrey-liventsev-ontarget**
Input markup refactoring - addonText and icon elements are flex element now
New Input option 'isClearable' added
Ndd new Input option 'highlightFilled' added

### CODE REVIEW

**Comment by johnny-five** [comment]
I, @cachapman, have checked:

### Style/Standards/Process

* [ ] If there is commented out code that should be deleted
* [ ] That the change implements what was requested in the [ticket](https://technekes.atlassian.net/browse/CCIBI-3739)
* [ ] For adequate test coverage for changed lines or critical code paths

### Code Quality

* [ ] Do the classes follow [CodeClimate 7 Patterns to Refactor](http://blog.codeclimate.com/blog/2012/10/17/7-ways-to-decompose-fat-activerecord-models/)
* [ ] If there are more idiomatic ways of using our frameworks/libraries
* [ ] Are there alternative implementations or refactors that increase readability/understandability and/or maintainability
* [ ] I can actually understand what the code is doing (if you're struggling to understand, then others may as well)
* [ ] This change is ready for production, if not should there should be a [feature flag](https://github.com/voormedia/flipflop)

### Security and Privacy

* [ ] Do the changes raise any security or privacy concerns

**Comment by cachapman** [comment]
Does this PR need to contain the text `Clear All` next to the cross icon for clearing the filters?

File: package-lock.json [hunk 1]
```diff
@@ -12065,9 +11972,9 @@
       }
     },
     "marked": {
-      "version": "0.7.0",
-      "resolved": "https://registry.npmjs.org/marked/-/marked-0.7.0.tgz",
-      "integrity": "sha512-c+yYdCZJQrsRjTPhUx7VKkApw9bwDkNbHUKo1ovgcfDjb2kc8rLuRbIFyXL5WOEUwzSSKo3IXpph2K6DqB/KZg==",
+      "version": "0.6.3",
```
Comment (cachapman) [inline package-lock.json:11975]: Downgrade?

File: package-lock.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@tkxs/cast-ui",
-  "version": "1.5.13",
+  "version": "1.5.1",
```
Comment (cachapman) [inline package-lock.json:3]: Downgrade?

Comment (andrey-liventsev-ontarget) [reply package-lock.json:3]: My mistake, I need to revert package-lock.json
Thanks for catching it

**Comment by andrey-liventsev-ontarget** [comment]
> Does this PR need to contain the text `Clear All` next to the cross icon for clearing the filters?

@cachapman This PR is only about new options to setup it from CES side
There will be another PR in CES repo to configure exact inputs

**Comment by cachapman** [comment]
> @cachapman This PR is only about new options to setup it from CES side
> There will be another PR in CES repo to configure exact inputs

@andrey-liventsev-ontarget is this what the `add-on` text is? If so, I think it would make sense to call it something like 'clear-filter-text' as 'add-on' could mean anything for anything haha. But if that's not it, then ignore me, but I'd like to know which variable sets that text then :-)

**Comment by andrey-liventsev-ontarget** [comment]
> > @cachapman This PR is only about new options to setup it from CES side
> > There will be another PR in CES repo to configure exact inputs
> 
> @andrey-liventsev-ontarget is this what the `add-on` text is? If so, I think it would make sense to call it something like 'clear-filter-text' as 'add-on' could mean anything for anything haha. But if that's not it, then ignore me, but I'd like to know which variable sets that text then :-)

@cachapman I added a new option **isClearable** to display X button if input value is not empty. Both **addonText** and **icon** were added a long time ago. I had to change markup to display them as flexed elements instead of using **position: absolute**. 
We are using **addonText** to display currencies and UOMs.

**Comment by andrey-liventsev-ontarget** [comment]
@cachapman I updated PR, please review the changes

**Pull request merged**